Tasks/GoTool/gotool.ts

```diff
@@ -0,0 +1,120 @@
+///<reference path="../../definitions/node.d.ts" />
+
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as tl from 'vsts-task-lib/task';
+import * as os from 'os';
+import * as path from 'path';
+
+let osPlat: string = os.platform();
+let osArch: string = os.arch();
+
+try {
+    tl.setResourcePath(path.join(__dirname, "task.json"));
+} catch (error) {
+    tl.setResult(tl.TaskResult.Failed, error);
+}
+
+async function run() {
+    try {
+        let versionSpec = tl.getInput('version', true);
+        await getGo(versionSpec);
+    }
+    catch (error) {
+        tl.setResult(tl.TaskResult.Failed, error);
+    }
+}
+
+async function getGo(version: string) {
+    // check cache
+    let toolPath: string;
+    toolPath = toolLib.findLocalTool('go', version);
+
+    if (!toolPath) {
+        // download, extract, cache
+        toolPath = await acquireGo(version);
+    }
+
+    setGoEnvironmentVariables(toolPath);
+
+    toolPath = path.join(toolPath, 'bin');
+    //
+    // prepend the tools path. instructs the agent to prepend for future tasks
+    //
+    toolLib.prependPath(toolPath);
+}
+
+
+async function acquireGo(version: string): Promise<string> {
+    //
+    // Download - a tool installer intimately knows how to get the tool (and construct urls)
+    //
+    version = toolLib.cleanVersion(version);
+    let fileName: string = getFileName(version);
+    let downloadUrl: string = getDownloadUrl(fileName);
+    let downloadPath: string = null;
+    try {
+        downloadPath = await toolLib.downloadTool(downloadUrl);
+    } catch (error) {
+        tl.debug(error);
+        throw (tl.loc("VersionNotSupported", version));
```

Comment:
if download fails for some reason we output version not supported?  What if download failure is environmental (network)?

Reply:
We don't have any way to determine what is the reason behind failure, other then parsing the error message, we don't want to enter in parsing business. So just for debugging we are logging actual error in debug.

Reply:
re-writing the error message with better and elaborate text

Reply:
Done
changes incorporated in https://github.com/Microsoft/vsts-tasks/pull/6307